filebeat/prospector/prospector.go

```diff
@@ -191,8 +213,9 @@ func (p *Prospector) startHarvester(state file.State, offset int64) error {
 			atomic.AddUint64(&p.harvesterCounter, ^uint64(0))
 			p.wg.Done()
 		}()
+
 		// Starts harvester and picks the right type. In case type is not set, set it to defeault (log)
-		h.Harvest()
+		h.Harvest(reader)
```

Comment:
why have harvester create reader in setup and finally pass it to Harvest? Isn't the reader stored in harvester?